### PR TITLE
Feature 2400 launch review period

### PR DESCRIPTION
**Pull request by mvolkmann**

### CODE REVIEW

File: web-ui/src/components/reviews/TeamReviews.jsx [hunk 1]
```diff
@@ -22,6 +22,10 @@ import {
 import {
   Alert,
   Button,
+  Card,
```
Comment (mvolkmann) [inline web-ui/src/components/reviews/TeamReviews.jsx:25]: New MUI components are needed for a new confirmation dialog.

File: web-ui/src/components/reviews/TeamReviews.jsx [hunk 2]
```diff
@@ -122,12 +127,13 @@ const ReviewStatus = {
 
 const TeamReviews = ({ onBack, periodId }) => {
   const { state, dispatch } = useContext(AppContext);
-  const history = useHistory();
```
Comment (mvolkmann) [inline web-ui/src/components/reviews/TeamReviews.jsx:125]: We don't need to use the History API to go back. We just need to call the `onBack` function that is passed in.

File: web-ui/src/components/reviews/TeamReviews.jsx [hunk 3]
```diff
@@ -611,7 +615,7 @@ const TeamReviews = ({ onBack, periodId }) => {
         },
         data: {
           ...period,
-          reviewStatus: ReviewStatus.AWAITING_APPROVAL
+          reviewStatus: ReviewStatus.OPEN
```
Comment (mvolkmann) [inline web-ui/src/components/reviews/TeamReviews.jsx:618]: When a review period is "launched", its status changes to OPEN, not AWAITING_APPROVAL.

File: web-ui/src/components/reviews/TeamReviews.jsx [hunk 4]
```diff
@@ -592,14 +599,11 @@ const TeamReviews = ({ onBack, periodId }) => {
       member => getReviewers(member).length > 0
     );
     if (!haveReviewers) return 'One or more members have no reviewer.';
+
     return null;
   };
 
-  const requestApproval = async () => {
```
Comment (mvolkmann) [inline web-ui/src/components/reviews/TeamReviews.jsx:598]: This is now done after a modal confirmation.

File: web-ui/src/components/reviews/TeamReviews.jsx [hunk 5]
```diff
@@ -621,6 +625,37 @@ const TeamReviews = ({ onBack, periodId }) => {
     }
   };
 
+  const requestApproval = async () => {
+    const msg = validateReviewPeriod(period);
+    setValidationMessage(msg);
+    if (msg) return;
+
+    const visibleIds = new Set(visibleTeamMembers().map(m => m.id));
```
Comment (mvolkmann) [inline web-ui/src/components/reviews/TeamReviews.jsx:633]: If there are no validation errors, we now display a modal dialog whose text depends on the number of unapproved review assignments.

File: web-ui/src/pages/ReviewsPage.jsx [hunk 6]
```diff
@@ -80,6 +80,7 @@ const ReviewPage = () => {
 
   const onPeriodSelected = useCallback(
     period => {
+      setSelectedPeriod(period);
```
Comment (mvolkmann) [inline web-ui/src/pages/ReviewsPage.jsx:83]: Setting this causes the UI to display the `TeamReviews` page.

File: web-ui/src/components/pulse/Pulse.test.jsx [hunk 7]
```diff
@@ -1,18 +1,17 @@
 import React from 'react';
-import renderer from 'react-test-renderer';
```
Comment (mvolkmann) [inline web-ui/src/components/pulse/Pulse.test.jsx:2]: Simplified this snapshot test.

File: web-ui/src/components/pulse/Pulse.test.jsx [hunk 8]
```diff
@@ -25,7 +24,7 @@ it('calls setComment', () => {
       setScore={() => {}}
     />
   );
-  const input = screen.getByTestId('comment-input').querySelector('input');
```
Comment (mvolkmann) [inline web-ui/src/components/pulse/Pulse.test.jsx:28]: The `input` was changed to a `textarea`.

File: web-ui/src/components/pulse/Pulse.test.jsx (hunk 7 above)
Comment (S78901) [reply web-ui/src/components/pulse/Pulse.test.jsx:2]: See changes to this file as well by @jackkeller on https://github.com/objectcomputing/check-ins/pull/2402

**Review by S78901 — APPROVED** [review]

**Review by vhscom — APPROVED** [review]
I think there may be a conflict following rebase but should be a straight-forward merge once the snapshots are regenerated. Code looks good.

**Pull request merged**